torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -119,7 +119,8 @@ ProcessGroupAgent::ProcessGroupAgent(
     workerIds_.emplace_back(std::move(tmpWorkerIds[rank]), rank);
   }
 
-  PythonRpcHandler::init();
+  // construct PythonRpcHandler singleton here
+  PythonRpcHandler::getInstance();
```

Comment:
We don't need to call this explicitly. Whenever the first `getInstance` method is encountered, the singleton would be created automatically.

Reply:
will it be better to explicitly create it in main thread? if multiple worker threads try to create it, they have to wait the lock